pkg/certificates/config.go

```
@@ -129,7 +130,95 @@ func ValidateComponentWithConfig(component string, config *RenewalConfig) error
 	return nil
 }
 
-// shouldProcessComponent checks if the specified component should be processed.
-func shouldProcessComponent(requestedComponent, targetComponent string) bool {
-	return requestedComponent == "" || requestedComponent == targetComponent
+func PopulateConfig(ctx context.Context, cfg *RenewalConfig, kubeClient kubernetes.Client, cluster *types.Cluster) error {
+
+	if len(cfg.ControlPlane.Nodes) > 0 {
+		return nil
+	}
+
+	controlPlaneIPs, err := getControlPlaneIPs(ctx, kubeClient, cluster)
+	if err != nil {
+		return fmt.Errorf("cluster is not reachable please provide control plane and/or external etcd IP addresses: %w", err)
+	}
+
+	cfg.ControlPlane.Nodes = controlPlaneIPs
+
+	etcdIPs, err := getEtcdIPs(ctx, kubeClient, cluster)
+	if err != nil {
+		fmt.Printf("Warning: Failed to get etcd IPs: %v\n", err)
```

please return an error from here.